src/main/java/datawave/microservice/authorization/util/AuthorizationsUtil.java

```diff
@@ -0,0 +1,177 @@
+        List<String> missingAuths = new ArrayList<>();
+        List<String> finalAuthsList = new ArrayList<>();
+        
+        for (String temp : requestedList) {
+            // user requested auth they don't have
+            if (!userList.contains(temp)) {
+                missingAuths.add(temp);
+            } else { // user requested auth they do have
+                finalAuthsList.add(temp);
+            }
+        }
+        // All auths requested are auths the user has, return downgraded string for auths
+        if (missingAuths.isEmpty()) {
+            return AuthorizationsUtil.buildAuthorizationString(Collections.singletonList(finalAuthsList));
+        } else {// missing auths.size() > 0; user requested auths they don't have
+            throw new IllegalArgumentException("User requested authorizations that they don't have. Missing: " + missingAuths + ", Requested: " + requested
+                            + ", User: " + userAuths);
+        }
+    }
+    
+    public static List<String> splitAuths(String requestedAuths) {
+        return Arrays.asList(Iterables.toArray(Splitter.on(',').omitEmptyStrings().trimResults().split(requestedAuths), String.class));
+    }
+    
+    public static Set<Authorizations> buildAuthorizations(Collection<? extends Collection<String>> userAuths) {
+        if (null == userAuths) {
+            return Collections.singleton(new Authorizations());
+        }
+        
+        HashSet<Authorizations> auths = Sets.newHashSet();
+        for (Collection<String> userAuth : userAuths) {
+            auths.add(new Authorizations(userAuth.toArray(new String[userAuth.size()])));
+        }
+        
+        return auths;
+    }
+    
+    public static String buildAuthorizationString(Collection<? extends Collection<String>> userAuths) {
+        if (null == userAuths) {
+            return "";
+        }
+        
+        HashSet<byte[]> b = new HashSet<>();
+        for (Collection<String> userAuth : userAuths) {
+            for (String string : userAuth) {
+                b.add(string.getBytes(StandardCharsets.UTF_8));
+            }
+        }
+        
+        return new Authorizations(b).toString();
+    }
+    
+    public static String buildUserAuthorizationString(DatawaveUserDetails currentUser) {
+        String auths = "";
+        if (currentUser != null) {
+            auths = new Authorizations(currentUser.getPrimaryUser().getAuths().toArray(new String[0])).toString();
+        }
+        return auths;
+    }
+    
+    public static Collection<Authorizations> minimize(Collection<Authorizations> authorizations) {
+        return AuthorizationsMinimizer.minimize(authorizations);
+    }
+    
+    public static Collection<? extends Collection<String>> prepareAuthsForMerge(Authorizations authorizations) {
+        return Collections.singleton(new HashSet<>(Arrays.asList(authorizations.toString().split(","))));
+    }
+}
```

Comment:
It seems worth adding tests for AuthorizationUtil.

Reply:
Sure, I'll copy the one from datawave-ws-common-util.  And it looks like Ivan has made changes to AuthorizationsUtil that I need to pull in as well, so I'll do that too.